flink-table/flink-table-common/src/main/java/org/apache/flink/table/data/ArrayData.java

```diff
@@ -163,7 +172,9 @@
 	 * @param pos position of the element to return
 	 * @param elementType the element type of the array
 	 * @return the element object at the specified position in this array data
+	 * @deprecated Use {@link #createElementGetter(LogicalType)} for a more efficient hot path.
 	 */
+	@Deprecated
```

Comment:
Could you create an issue to use `createElementGetter` to replace `get` utility and remove this method? I think we should provide a clean interface before release.